src/NuGet.Core/NuGet.Protocol/Converters/V3SearchResultsConverter.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using NuGet.Protocol.Model;
+
+namespace NuGet.Protocol.Converters
+{
+    internal class V3SearchResultsConverter : JsonConverter
+    {
+        private uint _take;
+        private JsonSerializer _newtonsoftConvertersSerializer;
+
+        public V3SearchResultsConverter(uint take, JsonSerializer newtonsoftConvertersSerializer)
+        {
+            _take = take;
+            _newtonsoftConvertersSerializer = newtonsoftConvertersSerializer;
+        }
+
+        public override bool CanWrite => false;
+
+        public override bool CanConvert(Type objectType) => objectType == typeof(V3SearchResults);
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (objectType != typeof(V3SearchResults))
+            {
+                throw new InvalidOperationException();
+            }
+
+            if (reader.TokenType != JsonToken.StartObject)
+            {
+                throw new JsonException("Expected StartObject, found " + reader.TokenType);
+            }
+
+            var searchResults = new V3SearchResults();
+
+            var finished = false;
+
+            while (!finished)
+            {
+                reader.Read();
+
+                switch (reader.TokenType)
+                {
+                    case JsonToken.PropertyName:
+                        switch ((string)reader.Value)
+                        {
+                            case "totalHits":
+                                long totalHits = long.TryParse(reader.ReadAsString(), out totalHits) ? totalHits : 0;
```

Comment:
I am assuming that the current code, that uses the default converter, throws.  It would be ideal to have the same exception message.  Same with the JsonException on line 58.

Reply:
Addressed